keanu-python/keanu/vertex/base.py

```diff
@@ -33,7 +33,7 @@ def __init__(self, val_or_ctor: Union[JavaMember, JavaObject], label: Optional[s
             val = typing_cast(JavaObject, val_or_ctor)
 
         super(Vertex, self).__init__(val)
-        if label is not None:
+        if label is not None and self.get_label() is None:
```

Comment:
when/how would `self.get_label()` not be `None`?

Reply:
When `label` is a required label for a vertex, it is set twice - by the constructor and here (this is currently how default labels are implemented).

For proxy vertices here it'll throw if you try to set a label after construction.

Reply:
Ah I get it. That's quite a subtle point (the distinction between the Java ctor and the Python ctor).